src/state/slices/userSlice.ts

```diff
@@ -0,0 +1,30 @@
+import { PayloadAction, createSlice } from "@reduxjs/toolkit";
+
+type UserState = {
+  value: User;
```

Comment:
Do we need this nested object behavior? Can we just ditch `value` and set the initial state to be a `user` object right away?

Same suggestion applies to `postSlice.ts`